ray-operator/controllers/ray/raycluster_controller_fake_test.go

```diff
@@ -1918,99 +1918,132 @@ func Test_RunningPods_RayContainerTerminated(t *testing.T) {
+					Name: "ray-head",
+					State: corev1.ContainerState{
+						Terminated: &corev1.ContainerStateTerminated{},
+					},
+				},
 			},
+			shouldDelete: false,
 		},
-	}
-	shouldDelete, _ = shouldDeletePod(pod, rayv1.HeadNode)
-	assert.False(t, shouldDelete)
-
-	// [Case 6]: The restart policy is set to `Never`, the Pod is not in a terminated state, and
-	// the Ray container has terminated. The expected behavior is that the controller will delete
-	// the Pod.
-	pod.Spec.RestartPolicy = corev1.RestartPolicyNever
-	pod.Status.Phase = corev1.PodRunning
-	pod.Status.ContainerStatuses = []corev1.ContainerStatus{
 		{
-			Name: "ray-head",
-			State: corev1.ContainerState{
-				Terminated: &corev1.ContainerStateTerminated{},
+			// The restart policy is `Never` and the Pod is in a terminate state.
+			// The expected behavior is that the controller will delete the Pod.
+			name:          "restartPolicy=Never, phase=PodFailed, shouldDelete=true",
+			restartPolicy: corev1.RestartPolicyNever,
+			phase:         corev1.PodFailed,
+			shouldDelete:  true,
+		},
+		{
+			// The restart policy is `Never` and the Pod terminated successfully.
+			// The expected behavior is that the controller will delete the Pod.
+			name:          "restartPolicy=Always, phase=PodSucceeded, shouldDelete=true",
+			restartPolicy: corev1.RestartPolicyNever,
+			phase:         corev1.PodSucceeded,
+			shouldDelete:  true,
+		},
+		{
+			// The restart policy is set to `Never`, the Pod is not in a terminated state, and
+			// the Ray container has not terminated. The expected behavior is that the controller will not
+			// delete the Pod.
+			name:          "restartPolicy=Always, phase=PodRunning, ray-head=running, shouldDelete=false",
+			restartPolicy: corev1.RestartPolicyNever,
+			phase:         corev1.PodRunning,
+			containerStatus: []corev1.ContainerStatus{
+				{
+					Name: "ray-head",
+					State: corev1.ContainerState{
+						Running: &corev1.ContainerStateRunning{},
+					},
+				},
 			},
+			shouldDelete: false,
 		},
+		{
+			// The restart policy is set to `Never`, the Pod is not in a terminated state, and
+			// the Ray container has terminated. The expected behavior is that the controller will delete
+			// the Pod.
+			name:          "restartPolicy=Always, phase=PodRunning, ray-head=terminated, shouldDelete=true",
+			restartPolicy: corev1.RestartPolicyNever,
+			phase:         corev1.PodRunning,
+			containerStatus: []corev1.ContainerStatus{
+				{
+					Name: "ray-head",
+					State: corev1.ContainerState{
+						Terminated: &corev1.ContainerStateTerminated{},
+					},
+				},
+			},
+			shouldDelete: true,
+		},
+	}
+	for _, testCase := range tests {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			pod.Spec.RestartPolicy = testCase.restartPolicy
+			pod.Status.Phase = testCase.phase
+			pod.Status.ContainerStatuses = testCase.containerStatus
+
+			shouldDelete, _ := shouldDeletePod(pod, rayv1.HeadNode)
+			assert.EqualValuesf(
```

Comment:
Is there a specific reason for using `assert.EqualValuesf` instead of assert.Equal? If not, perhaps we could use `assert.Equal` to maintain consistency.